datasafe-test-storages/src/main/java/de/adorsys/datasafe/teststorage/WithStorageProvider.java

```diff
@@ -269,15 +290,30 @@ private static void initS3() {
             return;
         }
 
-        amazonS3 = AmazonS3ClientBuilder.standard()
-        .withCredentials(new AWSStaticCredentialsProvider(
-                new BasicAWSCredentials(amazonAccessKeyID, amazonSecretAccessKey))
-        )
-        .withRegion(amazonRegion)
-        .build();
+        AmazonS3ClientBuilder amazonS3ClientBuilder = AmazonS3ClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(
+                        new BasicAWSCredentials(amazonAccessKeyID, amazonSecretAccessKey))
+                );
+
+        String[] buckets = amazonBucket.split(",");
+        if (buckets.length > 1) {
+            log.info("Using {} buckets:{}", buckets.length, amazonBucket);
+        }
+
+        if (StringUtils.isNullOrEmpty(amazonUrl)) {
+            amazonS3ClientBuilder = amazonS3ClientBuilder.withRegion(amazonRegion);
+            amazonMappedUrl = "s3://" + buckets[0] + "/" + bucketPath + "/";
+        } else {
+            amazonS3ClientBuilder = amazonS3ClientBuilder
+                    .withEndpointConfiguration(
+                            new AwsClientBuilder.EndpointConfiguration(amazonUrl, "US")
```

Comment:
Hardcoded "US" - it is better to use some env. var/constant

Reply:
ok